src/Control/Carrier/TaskPool.hs

```diff
@@ -87,39 +92,42 @@ stopWhenDone State{..} act = do
 worker :: (Has (Lift IO) sig m) => State m -> m ()
 worker st@State{..} = loop
   where
-
-  loop = join $ sendIO $ atomically $ stopWhenDone st $ do
-    queued <- readTVar stQueued
-    case queued of
-      [] -> retry
-      (x:xs) -> do
-        writeTVar stQueued xs
-        addRunning
-        pure $
-          mask $ \restore -> do
-            res <- try $ restore x
-            case res of
-              Left err -> do
-                _ <- sendIO $ atomically $ tryPutTMVar stError err
-                pure ()
-              Right () -> complete *> loop
-
-  addRunning :: STM ()
-  addRunning = modifyTVar stRunning (+1)
-
-  complete :: Has (Lift IO) sig m => m ()
-  complete = sendIO $ atomically $ modifyTVar stRunning (subtract 1) *> modifyTVar stCompleted (+1)
+    loop = join $
+      sendIO $
+        atomically $
+          stopWhenDone st $ do
+            queued <- readTVar stQueued
+            case queued of
+              [] -> retry
+              (x : xs) -> do
+                writeTVar stQueued xs
+                addRunning
+                pure $
+                  mask $ \restore -> do
+                    res <- try $ restore x
+                    case res of
+                      Left err -> do
+                        _ <- sendIO $ atomically $ tryPutTMVar stError err
+                        pure ()
+                      Right () -> complete *> loop
```

Comment:
Damn, so much for better treatment of whitespace.  But this is an edge case, after all.